cpp/src/arrow/compute/kernels/hash_aggregate_test.cc

```diff
@@ -692,6 +699,24 @@ TEST(GroupBy, Errors) {
                                         batch->GetColumnByName("group_id")}),
               Raises(StatusCode::NotImplemented,
                      HasSubstr("Direct execution of HASH_AGGREGATE functions")));
+
+  // Regression test for ARROW-14583: handle when no batches are
```

Comment:
Since this is not testing an error condition, should you create a separate test case for this?

Reply:
Yup, I split the test case.